### PR TITLE
Kernel+Userland: Expose partition metadata in SysFS, add printed info in lsblk about that metadata

### PR DESCRIPTION
**Pull request by supercomputer7**
Relies on #14661. Another nice step towards building a udev-like program in Serenity :)

### CODE REVIEW

**Review by SamuelBowman — COMMENTED** [review]
Looks good! I took a quick look through the code and nothing really stands out, though I'm not super familiar with kernel stuff, so maybe take that with a grain of salt. :^)

Looking ahead, to integrate this into a future PartitionEditor that does writes, we'll need a way to discover corresponding raw block device files. I would suggest adding a symlink that points to the device file in `/dev`, but that may be problematic since it'd be crossing a filesystem boundary and DevFS and SysFS could hypothetically be mounted to locations other than `/dev` and `/sys` respectively. Maybe we could just add a second block device file in `/sys/devices/storage/physical/LUN/` with the same major and minor numbers instead?

File: Kernel/FileSystem/SysFS/Subsystems/Devices/Storage/Logical/Partition/DeviceAttribute.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    virtual ErrorOr<size_t> read_bytes(off_t, size_t, UserOrKernelBuffer&, OpenFileDescription*) const override;
+    virtual ~PartitionDeviceAttributeSysFSComponent() {};
+
+    virtual StringView name() const override;
+
+protected:
+    ErrorOr<NonnullOwnPtr<KBuffer>> try_to_generate_buffer() const;
+    PartitionDeviceAttributeSysFSComponent(PartitionDeviceSysFSDirectory const& device, Type);
+    NonnullRefPtr<DiskPartition> m_device;
+    Type const m_type { Type::EndLBA };
```
[the remark this answers is not in the document]
Comment (supercomputer7) [reply Kernel/FileSystem/SysFS/Subsystems/Devices/Storage/Logical/Partition/DeviceAttribute.h:37]: Static Code analysis will warn us that we have an uninitialized primitive class member which is considered a bug, as far as I understand.

**Comment by supercomputer7** [comment]
> Looks good! I took a quick look through the code and nothing really stands out, though I'm not super familiar with kernel stuff, so maybe take that with a grain of salt. :^)
> 
> Looking ahead, to integrate this into a future PartitionEditor that does writes, we'll need a way to discover corresponding raw block device files. I would suggest adding a symlink that points to the device file in `/dev`, but that may be problematic since it'd be crossing a filesystem boundary and DevFS and SysFS could hypothetically be mounted to locations other than `/dev` and `/sys` respectively. Maybe we could just add a second block device file in `/sys/devices/storage/physical/LUN/` with the same major and minor numbers instead?

The kernel can't assume where userspace wants to mount `devtmpfs` nor `sysfs`. Therefore, the only sane thing to do is to expose a node for the major number and another for the minor number in each `/sys/devices/storage/physical/LUN/` directory. Device files should not be allowed to be created in the `sysfs` at all.

**Comment by supercomputer7** [comment]
> Looks good! I took a quick look through the code and nothing really stands out, though I'm not super familiar with kernel stuff, so maybe take that with a grain of salt. :^)
> 
> Looking ahead, to integrate this into a future PartitionEditor that does writes, we'll need a way to discover corresponding raw block device files. I would suggest adding a symlink that points to the device file in `/dev`, but that may be problematic since it'd be crossing a filesystem boundary and DevFS and SysFS could hypothetically be mounted to locations other than `/dev` and `/sys` respectively. Maybe we could just add a second block device file in `/sys/devices/storage/physical/LUN/` with the same major and minor numbers instead?

We could also (in relation to the previous comment of mine) just add a `device_identifier` symbolic link in each `/sys/devices/storage/physical/LUN/` to `/sys/dev/block/DEVICE_DIRECTORY_SYMLINK`. It is not a problem because unless you go and try to resolve this link endlessly, you won't have to encounter the `ELOOP` error...

**Comment by linusg** [comment]
This has conflicts

**Comment by supercomputer7** [comment]
Let's reopen this.

**Pull request closed without merging**